packages/ag-grid-theme/css/parts/ag-body.css

```diff
@@ -18,71 +27,134 @@ div[class*="ag-theme-salt"] .ag-row-selected:before {
+.ag-theme-salt-compact-dark .ag-ltr .ag-cell-inline-editing {
   padding: 0;
   /* When styling option corner='rounded', we don't want this to be flipped between rounded and not between editing and normal state */
   border-radius: 0;
 }
 
-div[class*="ag-theme-salt"] .ag-cell-inline-editing.editable-cell input[class^="ag-"] {
+.ag-theme-salt-light .ag-cell-inline-editing.editable-cell input[class^="ag-"],
+.ag-theme-salt-dark .ag-cell-inline-editing.editable-cell input[class^="ag-"],
+.ag-theme-salt-compact-light .ag-cell-inline-editing.editable-cell input[class^="ag-"],
+.ag-theme-salt-compact-dark .ag-cell-inline-editing.editable-cell input[class^="ag-"] {
   padding: 0 var(--salt-spacing-100);
 }
 
-div[class*="ag-theme-salt"] .editable-cell,
-div[class*="ag-theme-salt"] .editable-numeric-cell {
+.ag-theme-salt-light .editable-cell,
+.ag-theme-salt-light .editable-numeric-cell,
+.ag-theme-salt-dark .editable-cell,
+.ag-theme-salt-dark .editable-numeric-cell,
+.ag-theme-salt-compact-light .editable-cell,
+.ag-theme-salt-compact-light .editable-numeric-cell,
+.ag-theme-salt-compact-dark .editable-cell,
+.ag-theme-salt-compact-dark .editable-numeric-cell {
   outline: var(--salt-size-border) var(--salt-container-borderStyle) var(--salt-editable-borderColor);
   outline-offset: -1px;
 }
 
-div[class*="ag-theme-salt"] .ag-cell.numeric-cell,
-div[class*="ag-theme-salt"] .editable-numeric-cell {
+.ag-theme-salt-light .ag-cell.numeric-cell,
+.ag-theme-salt-light .editable-numeric-cell,
+.ag-theme-salt-dark .ag-cell.numeric-cell,
+.ag-theme-salt-dark .editable-numeric-cell,
+.ag-theme-salt-compact-light .ag-cell.numeric-cell,
+.ag-theme-salt-compact-light .editable-numeric-cell,
+.ag-theme-salt-compact-dark .ag-cell.numeric-cell,
+.ag-theme-salt-compact-dark .editable-numeric-cell {
   text-align: right;
 }
 
 /* Special case when user finish editing and click out side of the grid. */
-div[class*="ag-theme-salt"] .ag-cell.numeric-cell.ag-cell-inline-editing .ag-cell-editor input:not(:focus) {
+.ag-theme-salt-light .ag-cell.numeric-cell.ag-cell-inline-editing .ag-cell-editor input:not(:focus),
+.ag-theme-salt-dark .ag-cell.numeric-cell.ag-cell-inline-editing .ag-cell-editor input:not(:focus),
+.ag-theme-salt-compact-light .ag-cell.numeric-cell.ag-cell-inline-editing .ag-cell-editor input:not(:focus),
+.ag-theme-salt-compact-dark .ag-cell.numeric-cell.ag-cell-inline-editing .ag-cell-editor input:not(:focus) {
   text-align: right;
 }
 
-div[class*="ag-theme-salt"] .ag-has-focus .ag-cell.ag-cell-focus:not(.ag-cell-range-selected),
-div[class*="ag-theme-salt"] .ag-context-menu-open .ag-cell.ag-cell-focus:not(.ag-cell-range-selected),
-div[class*="ag-theme-salt"] .ag-cell-range-single-cell,
-div[class*="ag-theme-salt"] .ag-cell-range-single-cell.ag-cell-range-handle,
-div[class*="ag-theme-salt"] .ag-cell-focus:not(.ag-cell-range-selected):focus-within {
+.ag-theme-salt-light .ag-ltr .ag-cell-focus:not(.ag-cell-range-selected):focus-within,
```

Comment:
Changed cell focus selector to improve performance, followed ag grid css approach.